nntrainer/layers/embedding.cpp

```diff
@@ -104,17 +104,19 @@ void EmbeddingLayer::forwarding(bool training) {
     float *in_data = input_.getAddress(b * input_.getDim().getFeatureLen());
 
     for (unsigned int i = 0; i < in_length; ++i) {
-      if (in_data[i] < 0)
+      if (in_data[i] > in_dim) {
+        throw std::invalid_argument("input word index is greater then in_dim");
+      }
+
+      if (in_data[i] - 1 < 0)
```

Comment:
Does the indexing of data start from 0 or 1? If 0, then `in_data[i] == 0` is valid.

Reply:
It is padding and padding could be less or equal zero.

Reply:
`in_data` is the input data. I wonder how is it padding? can you please elaborate on the padding part?
I also wonder why the embedding layer needs padding.

Reply:
well.. the words is indexed as number. and the input is sentence. the maximum size of words is in_length.. and sometimes it is shorter than this. when it is short.. we need to fill out with padding. 
' Well done!' could be [ 1, 2, 0, 0] with in_length is 4.

Reply:
Then how making this check `in_data[i] == 0`?

Reply:
It also can be -1.

Reply:
Can you please add the possible values and what they denote as comments? BTW, what does -1 denote?

Reply:
I guess It depends how user set. As you know, index starts 0 or 1. right? If index starts 0, then it will -1. if it starts from 1, the padding will be 0. Nothing more.

Reply:
Then this condition will come out wrong if the actual index starts from 0, as this condition discards them too.